### PR TITLE
Update Swift dispatch to always be async

### PR DESCRIPTION
**Pull request by externl**
This PR updates the Swift dispatch code to always be async. I didn't do anything with the `amd` metadata. It's just ignored.

### CODE REVIEW

**Review by bernardnormier — COMMENTED** [review]
This PR highlights an issue with our new Swift setup: while the number of dispatch Tasks is typically large (limited by MaxDispatches), the number of actual thread is much smaller and a dispatch should not block synchronously - it should not onto a thread while waiting for an extended period of time.

For example, a dispatch should not hold onto a thread while:
 - ~waiting to acquire a lock~
 - waiting on a sync Ice API call that waits for some remote invocation to complete

In practice, it would make sense convert any "blocking" sync Ice API into an async API. For example Communicator::destroy can take a while => Communicator::destroy should be async.

See:
https://github.com/zeroc-ice/ice/blob/38da078632e5b1846efb46ee45197a052affb368/cpp/src/Ice/Instance.cpp#L1597
https://github.com/zeroc-ice/ice/blob/38da078632e5b1846efb46ee45197a052affb368/cpp/src/Ice/Instance.cpp#L1637

It's obviously impossible to fix in Swift unless Ice C++ provides an async API too.

A related question is what should a user do (if anything) if he needs to call a "bad" API that can block the thread. Starting a new Task doesn't help since it uses the same Executor. Start a new thread? Somehow fix the bad API by a proper async API (= which is what we should do for Ice).

**Review by bernardnormier — APPROVED** [review]

File: swift/test/Ice/ami/TestI.swift [hunk 1]
```diff
@@ -1,40 +1,87 @@
...
 
-class TestI: TestIntf {
-    var _batchCount: Int32
-    var _shutdown: Bool
-    var _lock = os_unfair_lock()
-    var _semaphore = DispatchSemaphore(value: 0)
+actor State {
+    var _batchCount: Int32 = 0
+    var _shutdown: Bool = false
     var _pending: CheckedContinuation<Void, Never>?
```
Comment (bernardnormier) [inline swift/test/Ice/ami/TestI.swift:11]: I don't find this code very elegant.

**Review by pepone — APPROVED** [review]

File: swift/test/Ice/admin/AllTests.swift [hunk 2]
```diff
@@ -16,15 +16,15 @@ class RemoteLoggerI: Ice.RemoteLogger {
         _helper = helper
     }
 
-    func `init`(prefix: String, logMessages: [Ice.LogMessage], current _: Ice.Current) throws {
+    func `init`(prefix: String, logMessages: [Ice.LogMessage], current _: Ice.Current) async throws {
```
[the remark this answers is not in the document]
Comment (externl) [reply swift/test/Ice/admin/AllTests.swift:19]: Yes, it's fine so long as we don't hold it while calling something asynchronously.

File: swift/test/Ice/binding/TestI.swift [hunk 3]
```diff
@@ -69,11 +69,11 @@ class RemoteObjectAdapterI: RemoteObjectAdapter {
         try _adapter.activate()
     }
 
-    func getTestIntf(current _: Ice.Current) throws -> TestIntfPrx? {
+    func getTestIntf(current _: Ice.Current) async throws -> TestIntfPrx? {
         return _testIntf
     }
 
-    func deactivate(current _: Ice.Current) throws {
+    func deactivate(current _: Ice.Current) async throws {
```
[the remark this answers is not in the document]
Comment (externl) [reply swift/test/Ice/binding/TestI.swift:76]: Yes, I think so so too.

**Pull request merged**